consensus/src/persistent_liveness_storage.rs

```diff
@@ -109,25 +109,26 @@ impl LedgerRecoveryData {
     ) -> Result<RootInfo> {
         // We start from the block that storage's latest ledger info, if storage has end-epoch
         // LedgerInfo, we generate the virtual genesis block
-        let (latest_commit_id, latest_ledger_info_sig) =
-            if self.storage_ledger.ledger_info().ends_epoch() {
-                let genesis =
-                    Block::make_genesis_block_from_ledger_info(self.storage_ledger.ledger_info());
-                let genesis_qc = QuorumCert::certificate_for_genesis_from_ledger_info(
-                    self.storage_ledger.ledger_info(),
-                    genesis.id(),
-                );
-                let genesis_ledger_info = genesis_qc.ledger_info().clone();
-                let genesis_id = genesis.id();
-                blocks.push(genesis);
-                quorum_certs.push(genesis_qc);
-                (genesis_id, genesis_ledger_info)
-            } else {
-                (
-                    self.storage_ledger.ledger_info().consensus_block_id(),
-                    self.storage_ledger.clone(),
-                )
-            };
+        let ends_epoch = self.storage_ledger.ledger_info().ends_epoch();
+        let blocks_empty = blocks.is_empty();
```

Comment:
Perhaps you should add a comment why we consider to have an empty block the same as an end epoch.

Reply:
updated